source/server/guarddog_impl.cc

```diff
@@ -38,6 +49,29 @@ GuardDogImpl::GuardDogImpl(Stats::Scope& stats_scope, const Server::Configuratio
       watchdog_megamiss_counter_(stats_scope.counterFromStatName(
           Stats::StatNameManagedStorage("server.watchdog_mega_miss", stats_scope.symbolTable())
               .statName())),
+      event_to_callbacks_([](const Server::Configuration::Main& config) -> EventToCallbackMap {
+        EventToCallbackMap map = {
+          {WatchDogEvent::Watchdog_WatchdogAction_WatchdogEvent_KILL, {}},
+          {WatchDogEvent::Watchdog_WatchdogAction_WatchdogEvent_MULTIKILL, {}},
+          {WatchDogEvent::Watchdog_WatchdogAction_WatchdogEvent_MISS, {}},
+          {WatchDogEvent::Watchdog_WatchdogAction_WatchdogEvent_MEGAMISS, {}},
+        };
+
+        Configuration::GuardDogActionFactoryContext context;
+        
+        const auto& actions = config.wdActions();
+        for(const auto& action : actions) {
+          if(action.event() == WatchDogEvent::Watchdog_WatchdogAction_WatchdogEvent_UNKNOWN) {
+            ENVOY_LOG_MISC(error, "WatchDogAction specified with UNKNOWN event");
+          } else {
+            // Get factory and add the created cb
+            auto& factory = Config::Utility::getAndCheckFactoryByName<Configuration::GuardDogActionFactory>(action.name());
```

Comment:
Oh, this is a full blown factory for the guard dog actions. Why not use `TypedExtensionConfig` then and make the actions fully opaquely configurable?

Reply:
Will go with that, thanks for the pointer.